config/config.go

```diff
@@ -93,6 +95,37 @@ func RemoveConfig() error {
 	return os.Remove(ConfigPath())
 }
 
+func (cfg *Config) Client() (*credhub.CredHub, error) {
```

Comment:
Hi Bruce. My main motivation for putting this function inside the config module was so that it can be reused. For simple use, configuring `credhub.New` takes more time to figure out than does configuring the structs in the config module. Exporting this function from the config module means we can use it elsewhere.

But I do agree there's no need for this function to go on the actual `Config` struct. I'll move this to be a standalone function in the config module. Happy to talk more about it.

Reply:
> Exporting this function from the config module means we can use it elsewhere.

I guess I don't really get this argument. IMHO only the `main` should know about the config, use it to create objects, and pass them down the call stack. Inner functions/components digging for stuff into the config, in our experience, breaks the principle of least privilege. Would you agree with this?